src/core/job.c

```diff
@@ -509,6 +509,93 @@ static void job_change_type(Job *j, JobType newtype) {
         j->type = newtype;
 }
 
+_pure_ static const char* job_get_begin_status_message_format(Unit *u, JobType t) {
+        const char *format;
+        const UnitStatusMessageFormats *format_table;
+
+        assert(u);
+        assert(IN_SET(t, JOB_START, JOB_STOP, JOB_RELOAD));
+
+        if (t != JOB_RELOAD) {
+                format_table = &UNIT_VTABLE(u)->status_message_formats;
+                if (format_table) {
```

Comment:
This is a pointer to a field, so it cannot be zero. This whole variable can go away, and `UNIT_VTABLE(u)->status_message_formats->starting_stopping[t == JOB_STOP]` can be accessed directly.